tests/test_redcap_bp_utils.py

```diff
@@ -66,3 +67,265 @@ def study(self, dash_db):
+
+    def test_exception_raised_if_record_not_found_on_server(
+            self, mock_http, mock_monitor, det, records):
+        mock_http.return_value = self.mock_redcap_export([])
+        with pytest.raises(dashboard.exceptions.RedcapException):
+            rc_utils.create_from_request(det)
+
+    def test_adds_redcap_record_to_session(
+            self, mock_http, mock_monitor, det, records):
+        mock_http.return_value = self.mock_redcap_export()
+        session = dashboard.models.Session.query.get((self.session[:-3], 1))
+        assert session.redcap_record is None
+
+        rc_utils.create_from_request(det)
+
+        assert session.redcap_record is not None
+        assert session.redcap_record.record.comment == self.comment
+
+    def test_adds_session_if_doesnt_exist(
+            self, mock_http, mock_monitor, det, records):
+        mock_http.return_value = self.mock_redcap_export()
+
+        timepoint = self.session[:-3]
+        session = dashboard.models.Session.query.get((timepoint, 1))
+        if session:
+            session.delete()
+        assert dashboard.models.Session.query.get((timepoint, 1)) is None
+        rc_utils.create_from_request(det)
+
+        session = dashboard.models.Session.query.get((timepoint, 1))
+        assert session is not None
+
+    def test_updates_redcap_record_if_updates_on_server(
+            self, mock_http, mock_monitor, det, records):
+        mock_http.return_value = self.mock_redcap_export()
+
+        # Add a redcap record
+        rc_utils.create_from_request(det)
+
+        # Modify the comment on the 'server'
+        new_comment = "Scan was NOT ok."
+        new_record = self.redcap_records[0].copy()
+        new_record[self.comment_field] = new_comment
+        mock_http.return_value = self.mock_redcap_export([new_record])
+
+        # Resend the data entry trigger
+        rc_utils.create_from_request(det)
+
+        session = dashboard.models.Session.query.get((self.session[:-3], 1))
+        assert session is not None
+        assert session.redcap_record.record.comment == new_comment
+
+    def test_sets_event_id_of_redcap_record(
+            self, mock_http, mock_monitor, det, records):
+        mock_http.return_value = self.mock_redcap_export()
+        rc_utils.create_from_request(det)
+
+        session = dashboard.models.Session.query.get((self.session[:-3], 1))
+        assert session.redcap_record is not None
+        db_record = session.redcap_record.record
+
+        assert db_record.event_id == self.event_ids[self.event_name]
+
+    def test_doesnt_fail_if_config_event_ids_not_set(
+            self, mock_http, mock_monitor, det, records):
+        mock_http.return_value = self.mock_redcap_export()
+
+        # Delete the event_ids from config
+        rc = dashboard.models.RedcapConfig.query.get(1)
+        rc.event_ids = None
+        dashboard.models.db.session.add(rc)
+        dashboard.models.db.session.commit()
+
+        assert not rc.event_ids
```

Comment:
```suggestion
        assert rc.event_ids is None
```

Would making this more explicit work ok? Or will `rc.event_ids` not return a null type if unset